src/System.Text.Http/System/Text/Http/Obsolete/HttpRequestParser.cs

```diff
@@ -65,11 +65,15 @@ public static HttpRequestSingleSegment Parse(ReadOnlySpan<byte> bytes)
         }
     }
 
-    public struct HttpRequestReader
+    public ref struct HttpRequestReader
     {
-        static readonly Utf8String s_Http1_0 = new Utf8String("HTTP/1.0");
-        static readonly Utf8String s_Http1_1 = new Utf8String("HTTP/1.1");
-        static readonly Utf8String s_Http2_0 = new Utf8String("HTTP/2.0");
+        static readonly byte[] b_Http1_0 = new Utf8String("HTTP/1.0").CopyBytes();
+        static readonly byte[] b_Http1_1 = new Utf8String("HTTP/1.1").CopyBytes();
+        static readonly byte[] b_Http2_0 = new Utf8String("HTTP/2.0").CopyBytes();
+
+        static Utf8String s_Http1_0 => new Utf8String(b_Http1_0);
```

Comment:
yes, properties/methods can return spans or ref structs in general. 
The fields are not allowed, unless they are instance fields in a ref struct.